pytorch_lightning/profiler/profilers.py

```diff
@@ -55,23 +43,47 @@ def start(self, action_name: str) -> None:
     def stop(self, action_name: str) -> None:
         """Defines how to record the duration once an action is complete."""
 
-    def setup(
+    @abstractmethod
+    def summary(self) -> str:
+        """Create profiler summary in text format."""
+
+    @abstractmethod
+    def setup(self) -> None:
+        """Execute arbitrary pre-profiling set-up steps as defined by subclass."""
```

Comment:
should setup take **kwargs?

Reply:
Do you see any reasons for it ?

Reply:
The base class overrides this differently by adding more args

Reply:
@carmocca Let's add kwargs there and below.